recipes/boost/all/conanfile.py

```diff
@@ -188,6 +222,21 @@ def config_options(self):
             if "without_{}".format(opt_name) not in self.options:
                 raise ConanException("{} has the configure options {} which is not available in conanfile.py".format(self._dependency_filename, opt_name))
 
+        # nowide requires a c++11-able compiler + movable std::fstream: change default to not build on compiler with too old default c++ standard or to low compiler.cppstd
+        # json requires a c++11-able compiler: change default to not build on compiler with too old default c++ standard or to low compiler.cppstd
+        if self.settings.compiler.cppstd:
+            if not tools.valid_min_cppstd(self, 11):
+                self.options.without_fiber = True
+                self.options.without_nowide = True
+                self.options.without_json = True
+        else:
+            version_cxx11_standard_json = self._min_compiler_version_default_cxx11
+            if version_cxx11_standard_json:
+                if tools.Version(self.settings.compiler.version) < version_cxx11_standard_json:
+                    self.options.without_fiber = True
+                    self.options.without_json = True
+                    self.options.without_nowide = True
```

Comment:
I think if we change the value of some option, we should notify user with a warning.

Reply:
This is in `config_options`.
When you assign a value to `self.options.xxx`, then you actually change the default value for that option.

The evaluation order of an option is:
1. Assign self.option.xxx from self.default_options
2. Run `config_options`
3. Assign option values from profile + cli arguments
4. Run `configure`

So the user can always override.

For e.g. fiber, there is no difference in behavior.
Before, with an unsupported compiler, even when you set `without_fiber=False`, fiber would not be built.

Reply:
Thanks for the explanation 👍